common/helpers.js

```diff
@@ -200,6 +237,25 @@ function getMd5Checksum(checksums) {
     return (checksums.find((e) => e.type === 'md5') || {}).checksum;
 }
 
+/**
+ * Transforms the DOS or DRS checksum array into a map where `type` is key and `checksum` is value
+ *
+ * @param {Object[]} checksumArray The checksum of the drs object
+ * @param {string} checksums[].checksum The hex-string encoded checksum for the data
+ * @param {string} checksums[].type The digest method used to create the checksum
+ * @returns {Object} The checksum map as an object
+ */
+function getHashesMap(checksumArray) {
+    /*
+        if there are more than 1 checksum for same type of hash, the last hash value in the checksums array for that
+        type will be taken
+     */
+    return checksumArray.reduce(function(map, obj){
+        map[obj.type] = obj.checksum;
+        return map;
+    }, {});
```

Comment:
Beautiful. TIL. 😮👏🥂

Btw, still wish there were _some way_ of adding types to JavaScript/ECMAScript... if only someone out there had come up with something...